src/FEValues/GeometryMapping.jl

```diff
@@ -121,15 +121,32 @@ end
     return Tensor{3,dim,promote_type(Tx,TM)}
 end =# # PR798
 
-@inline function calculate_mapping(::GeometryMapping{0}, q_point, x)
+@inline function calculate_mapping(gip::VectorizedInterpolation, ξ::Vec, cell_coordinates::AbstractVector{<:Vec})
+    return calculate_mapping(gip.ip, ξ, x)
+end
+
+@inline function calculate_mapping(gip::ScalarInterpolation, ξ::Vec, cell_coordinates::AbstractVector{<:Vec})
+    n_basefuncs = getnbasefunctions(gip)
+    @assert length(cell_coordinates) == n_basefuncs
```

Comment:
Driveby: 
```suggestion
    @boundscheck checkbounds(cell_coordinates, Base.OneTo(n_basefuncs))
```
To be safe for `AbstractVector` and to allow escape by `@inbounds` from outside?
Could also be reasonable to check for the same length to catch user bugs?

Also good to note that the `calculate_mapping` function that I wrote above is unsafe regarding inbounds. 
It would be required to either remove the unchecked `@inbounds` and add a `@propagate_inbounds`, or add something like an `unsafe_calculate_mapping` which assumes inbounds are ok to use. But the performance was tricky and easily disturbed...

Reply:
Should I add the note here or rather open an issue?